test/test_train_mp_imagenet.py

```diff
@@ -200,16 +205,30 @@ def train_imagenet():
       num_steps_per_epoch=num_training_steps_per_epoch,
       summary_writer=writer)
   loss_fn = nn.CrossEntropyLoss()
+  if FLAGS.amp:
+    scaler = GradScaler()
 
   def train_loop_fn(loader, epoch):
     tracker = xm.RateTracker()
     model.train()
     for step, (data, target) in enumerate(loader):
-      optimizer.zero_grad()
-      output = model(data)
-      loss = loss_fn(output, target)
-      loss.backward()
-      xm.optimizer_step(optimizer)
+      if FLAGS.amp:
+        with autocast():
+          output = model(data)
+          loss = loss_fn(output, target)
+
+        scaler.scale(loss).backward()
+        gradients = xm._fetch_gradients(optimizer)
+        xm.all_reduce('sum', gradients, scale=1.0 / xm.xrt_world_size())
+        scaler.step(optimizer)
+        scaler.update()
+        xm.mark_step()
+      else:
```

Comment:
this no longer has `optimizer.zero_grad()` like in the old version. Is that OK to leave out?